### ISSUE TITLE
Haskell extension reports an Uncaught Error, log file is empty

### ISSUE THREAD

**Issue by L7R7** [haskell/vscode-haskell#517]
Recently, the Haskell support in VS Code doesn't work anymore for any of my projects. I assume it is related with the latest version of the extension or the HLS.

In the extension inside VS Code it says that there is an uncaught error (see screenshot below).

### Your environment

OS version: Ubuntu 21.04
VS Code version: 1.62.3

### Steps to reproduce

It's the same behavior for all my projects. The error description in the extension tab looks like that:

![image](https://user-images.githubusercontent.com/16477399/145095209-507f9418-aabf-4d4f-b83c-4786259586cd.png)

* `/home/leo/dev/aoc/aoc2021` is the root directory of my project
* There is no `hls.log` file at the described location, but there is one in my home directory (`~/hls.log`) which is empty

### HLS logs

<details>
<summary>
log output:
</summary>

```
2021-12-07 19:32:37.1360000 [client] INFO Writing client log to file /home/leo/dev/aoc/aoc2021/~/hls.log
```
</details>

**Comment by L7R7** [comment]
Well, shortly after creating this issue, I had the idea to change the log file location in the extension settings. It was set to `~/hls.log`. I changed it to `hls.log` and now it works fine.

DId something change here? I never changed the settings before so it's a bit surprising that it stops working

**Comment by jneira** [comment]
HI, thanks for opening the issue. 

> DId something change here? I never changed the settings before so it's a bit surprising that it stops working

I think so, iirc the extension does an alias expansion of paths and maybe some of latest changes has affected it, needs to be investigated

**Comment by jneira** [comment]
Hi, i've identified the change which broke your use case: https://github.com/haskell/vscode-haskell/commit/2aa1f64db58ec476cd69aa35a0f7e2ce0c61efe2
I assumed incorrectly that `path.resolve(currentWorkingDir, '~/hls.log') == '~/hls.log'`

Sorry for the inconveniences, will fix it asap

**Comment by jneira** [comment]
@L7R7 hi! i've prepared a fix with #518, could you got the chance of test the pr vsix to check if it fix the issue in your env (setting the log path again to `~/hls.log`)?

https://github.com/haskell/vscode-haskell/suites/4605389050/artifacts/124395850

**Comment by L7R7** [comment]
@jneira I can confirm it works for me with the updated version.

Thanks for the fast response!
